packages/filter-select/src/FilterSelect/TriggerButton/TriggerButtonBase.tsx

```diff
@@ -0,0 +1,49 @@
+import React from "react"
+import chevronDown from "@kaizen/component-library/icons/chevron-down.icon.svg"
+import { Icon } from "@kaizen/component-library"
+import classNames from "classnames"
+import clear from "@kaizen/component-library/icons/clear.icon.svg"
+
+import { useMenuTriggerContext } from "../provider/MenuTriggerProvider"
+import styles from "./TriggerButton.scss"
+
+export interface TriggerButtonBaseProps {
+  children: React.ReactNode
+  isRemovalbe?: boolean
+  onRemove?: () => void
```

Comment:
made it union optional type for now and will let it sit for any feedback :) : 
```
interface RemovableProps {
  isRemovable: true
  onRemove: () => void
}

interface RemovableUndefinedProps {
  isRemovable?: false
  onRemove?: never
}

export type TriggerButtonBaseProps = {
  children: React.ReactNode
  hasSelectedValues?: boolean
} & (RemovableProps | RemovableUndefinedProps)
```
e.g. 
```
// error 
{
  isRemovable: true,
   /* expect onRemove: () => void*/
}

{
  isRemovable: false,
  onRemove: () => {}  /* expect onRemove: undefined*/
}

// accpetable 
{
  isRemovable: true,
  onRemove: () => {...}
}

{
  isRemovable: false,
}
```

Reply:
I've just posted to check what my team thinks. One other idea popped into my head, but I don't know if it's feasible. Given how the rest of this component is designed from a consuming perspective, is it possible that the remove button becomes its own thing instead of driven by a prop on the main component?
```
<FilterMultiSelect>
...
</FilterMultiSelect>
<FilterMultiSelect.DeleteButton />
```

Reply:
Given the props staying on < FilterMultiSelect.TriggerButton /> so maybe something like this? :

`<FilterMultiSelect.TriggerButton />`
`<FilterMultiSelect.RemovableTriggerButton />` // include onRemove on the props

Reply:
and just realised it is what Cass suggested in the slack post :)